### PR TITLE
Support overriding server args

### PR DESCRIPTION
**Pull request by mig4**
Towards: https://github.com/giantswarm/giantswarm/issues/18571

Add support for overriding server args through Helm values.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Review by QuentinBisson — APPROVED** [review]

**Review by QuentinBisson — CHANGES_REQUESTED** [review]
Actually, I think it would make more sense for the arg to be set in giantswarm/config

**Comment by mig4** [comment]
> Actually, I think it would make more sense for the arg to be set in giantswarm/config

What's our convention for this? Have a `Values.server.args` and allow specifying anything, or explicitly `Values.server.disable_user_metrics` bool and if true then add the flag?

**Comment by QuentinBisson** [comment]
I would think we go with `Values.server.args` to provide more flexibility

**Comment by JosephSalisbury** [comment]
Counter argument: would we ever want to store this information? Why allow it to be configurable?

**Comment by mig4** [comment]
> Counter argument: would we ever want to store this information? Why allow it to be configurable?

I imagine this is for if customers wanted to install this through App platform, right? I don't think we would use it.

**Comment by JosephSalisbury** [comment]
Oh sure - I agree with y'all then :D

File: helm/macropower-analytics-panel-server/templates/deployment.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ spec:
       - name: {{ include "name" . }}
         image: "{{ .Values.registry.domain }}/{{ .Values.server.image.name }}:{{ .Values.server.image.tag }}"
         imagePullPolicy: {{ .Values.server.image.pullPolicy }}
+        args: {{ .Values.server.args }}
```
Comment (mig4) [inline helm/macropower-analytics-panel-server/templates/deployment.yaml:36]: @QuentinBisson is this right? Do I need to do something special here to inject an array?

**Comment by mig4** [comment]
Done moving out the arg, so this goes with: https://github.com/giantswarm/config/pull/345

I'll change the title and description of this PR to just _adding args_ or so.

**Review by QuentinBisson — APPROVED** [review]

File: helm/macropower-analytics-panel-server/Chart.yaml [hunk 2]
```diff
@@ -9,5 +9,5 @@ sources:
 - https://github.com/MacroPower/macropower-analytics-panel/tree/master/server
 version: [[ .Version ]]
 annotations:
-  config.giantswarm.io/version: 1.x.x
+  config.giantswarm.io/version: "analytics-server-disable-user-metrics"
```
Comment (mig4) [inline helm/macropower-analytics-panel-server/Chart.yaml:12]: I tried to test this with the [associated change in config](https://github.com/giantswarm/config/pull/345) but I couldn't get this to work. Regardless of whether the syntax is right in the deployment template, this `config.giantswarm.io/version` setting wouldn't be respected, i.e. config-controller generates the relevant ConfigMap from released version as it does not include the field added in [analytics-server-disable-user-metrics](https://github.com/giantswarm/config/tree/analytics-server-disable-user-metrics) branch.

File: helm/macropower-analytics-panel-server/templates/deployment.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/macropower-analytics-panel-server/templates/deployment.yaml]: done

File: helm/macropower-analytics-panel-server/Chart.yaml (hunk 2 above)
Comment (TheoBrigitte) [reply helm/macropower-analytics-panel-server/Chart.yaml:12]: Looks like both the repo and the config branch have to be the same name. Closing this PR in favor of https://github.com/giantswarm/macropower-analytics-panel-server-app/pull/20

Comment (mig4) [reply helm/macropower-analytics-panel-server/Chart.yaml:12]: OK, that's surprising, or maybe even a bug. Gotta ask you about it when you're back, feels like we should at least document it in https://intranet.giantswarm.io/docs/dev-and-releng/configuration-management/#how-can-i-test-giantswarmconfig-changes-without-making-a-release

**Pull request closed without merging**